saffron/src/encoding.rs

```diff
@@ -0,0 +1,163 @@
+//! This file handles bytes <-> scalar conversions for Saffron.
+//! Unless specified in the function's name, conversions are made over
+//! `F::MODULUS_BIT_SIZE / 8` bytes (31 bytes for Pallas & Vesta). Functions
+//! that convert over `F::size_in_bytes()` are suffixed with `_full` (this size
+//! is 32 bytes for Pallas & Vesta fields elements)
+
+use ark_ff::{BigInteger, PrimeField};
+use o1_utils::FieldHelpers;
+
+/// The size in bytes of the full representation of a field element (32 for
+/// Pallas & Vesta)
+pub fn encoding_size_full<F: PrimeField>() -> usize {
+    F::size_in_bytes()
+}
+
+/// The number of bytes that can be fully represented by a scalar (31 for
+/// Pallas & Vesta)
+pub fn encoding_size<F: PrimeField>() -> usize {
+    (F::MODULUS_BIT_SIZE / 8) as usize
+}
+
+// For injectivity, you can only use this on inputs of length at most
+// 'F::MODULUS_BIT_SIZE / 8', e.g. for Pallas & Vesta this is 31.
+/// Converts `bytes` into a field element ; `bytes` length can be arbitrary.
+pub fn encode<F: PrimeField>(bytes: &[u8]) -> F {
+    F::from_be_bytes_mod_order(bytes)
+}
+
+/// Returns the `F::size_in_bytes()` representation of `x`
```

Comment:
Nit: I would mention that the chosen representation of the implementation is the decimal representation of the field element in big endian.